flash/core/utilities/flash_cli.py

```diff
@@ -211,22 +219,33 @@ def add_arguments_to_parser(self, parser) -> None:
 
     def add_subcommand_from_function(self, subcommands, function, function_name=None):
         subcommand = ArgumentParser()
-        datamodule_function = class_from_function(drop_kwargs(function), return_type=self.local_datamodule_class)
-        subcommand.add_class_arguments(datamodule_function, fail_untyped=False)
         if self.legacy:
+            datamodule_function = class_from_function(drop_kwargs(function), return_type=self.local_datamodule_class)
+            subcommand.add_class_arguments(datamodule_function, fail_untyped=False)
             input_transform_function = class_from_function(drop_kwargs(self.local_datamodule_class.input_transform_cls))
             subcommand.add_class_arguments(
                 input_transform_function,
                 fail_untyped=False,
                 skip=get_overlapping_args(datamodule_function, input_transform_function),
             )
-        else:
-            base_datamodule_function = class_from_function(drop_kwargs(self.local_datamodule_class))
+        elif get_kwarg_name(function) == "data_module_kwargs":
+            datamodule_function = class_from_function(function, return_type=self.local_datamodule_class)
             subcommand.add_class_arguments(
-                base_datamodule_function,
+                datamodule_function,
                 fail_untyped=False,
-                skip=get_overlapping_args(datamodule_function, base_datamodule_function),
+                skip={
+                    "self",
+                    "train_dataset",
```

Comment:
why skip this?

Reply:
These are the arguments used by the methods (e.g. `from_*`) internally to create the DataModule. So if you don't skip you get a collision where they are also provided by jsonargparse